### ISSUE TITLE
Lint that replaces a destructive assignment of one field from a tuple with an index-based value

### ISSUE THREAD

**Issue by MarkTanashchuk** [rust-lang/rust-clippy#7530]
### What it does

This lint recommends avoiding destructuring the tuple if only one of its values is used and the rest are marked `_` and, instead, getting the value directly by index.

### Categories (optional)

- Kind: `clippy::style`

<br class="Apple-interchange-newline">

*What is the advantage of the recommended code over the original code*

It's shorter and easier to read.
The difference is especially noticeable in tuples of size 4 and larger.

### Drawbacks

There shouldn't be, but I can be wrong.

### Example

```rust
let (variableName, _)     = some_tuple_double;
let (_, variableName2, _) = some_tuple_triple;
```

Could be written as:

```rust
let variableName  = some_tuple_double.1;
let variableName2 = some_tuple_triple.2;
```

**Comment by MarkTanashchuk** [comment]
Although, in fact, with such a warning the rust compiler will not warn the programmer that the number of tuple elements has changed (as it would in the case of destructuring due to an inappropriate pattern), and the programmer may forget to change the index.

However, the rest pattern(
`let (.., variableName) = ("One", "Two", "Three");`
) is used widely, and programmers are relatively okay with it, so the danger of the programmer forgetting about the index should be given some more thought.
